source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,100 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+
+/**
+ * A hook for displaying link tooltips which can be used to visually
+ * enable users to click a link (without having to press down Ctrl/Cmd)
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (cm) => {
+  cm.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+
+    // Initialise displayed attributes to 'loading'
+    let title = 'loading'
+    let content = 'loading'
+    let wordCount = 'loading'
+    let time = 'loading'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false, // No arrow for these tooltips
+      onHidden (instance) {
+        instance.destroy() // Destroy the tippy instance.
+      }
+    })
+
+    // Find the file's absolute path
+    ipcRenderer.invoke('application', { command: 'file-path-find', payload: a.innerText })
+      .then((filepath) => {
+        // If the file is found
+        if (filepath !== 'Not Found') {
+          // Retrieve the file contets
+          ipcRenderer.invoke('application', { command: 'get-file-contents', payload: filepath })
+            .then((descriptorWithContent) => {
+              // Remove html from content so html is displayed normally. (delete html tags)
+              descriptorWithContent.content = descriptorWithContent.content.replace(/(<([^>]+)>)/gi, '')
+
+              // Get the contents of the file such as:
+              // 4 lines of 50
+              content = descriptorWithContent.content.substring(0, 50)
+              if (descriptorWithContent.content.length >= 50) {
+                content += '\n' + descriptorWithContent.content.substring(50, 100)
+                if (descriptorWithContent.content.length >= 100) {
+                  content += '\n' + descriptorWithContent.content.substring(100, 150)
+                  if (descriptorWithContent.content.length >= 150) {
+                    content += '\n' + descriptorWithContent.content.substring(150, 200)
+                    // Add an elipsis if there is more left
+                    if (descriptorWithContent.content.length >= 200) {
+                      content += '...'
```

Comment:
… what are you doing here? This looks both unnecessarily tedious, has to many nested `if`-statements and I'm not sure why we'd need this in the first place …?

Reply:
This was because if I just assumed there was enough content and added newlines, I would get an extra space at the end oof the content. I will refactor this so it's a loop with just a scaling conditional check each time though.